### PR TITLE
sdit-450 add duplicate error handling to visits

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt [hunk 1]
```diff
@@ -53,6 +53,16 @@ class IncentiveMappingResource(private val mappingService: IncentiveMappingServi
...
+        responseCode = "409",
+        description = "Indicates a duplicate incentive has been rejected. If Error code = 1409 the body will return a DuplicateErrorResponse",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = DuplicateMappingErrorResponse::class),
+          ),
+        ],
+      ),
     ],
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:66]: moving response to the create endpoint (was incorrectly on GET)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt [hunk 2]
```diff
@@ -53,6 +53,16 @@ class IncentiveMappingResource(private val mappingService: IncentiveMappingServi
         description = "Unauthorized to access this endpoint",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
       ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Indicates a duplicate incentive has been rejected. If Error code = 1409 the body will return a DuplicateErrorResponse",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:58]: error code `1409`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/VisitMappingResource.kt [hunk 3]
```diff
@@ -54,6 +55,16 @@ class VisitMappingResource(private val mappingService: VisitMappingService) {
         description = "Unauthorized to access this endpoint",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
       ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Indicates a duplicate visit has been rejected. If Error code = 1409 the body will return a DuplicateErrorResponse",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/VisitMappingResource.kt:60]: `1409`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/SentencingMappingResource.kt [hunk 4]
```diff
@@ -58,6 +58,16 @@ class SentencingMappingResource(private val mappingService: SentencingMappingSer
         description = "Unauthorized to access this endpoint",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
       ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Indicates a duplicate sentencing adjustment mapping has been rejected. If Error code = 409 the body will return a DuplicateErrorResponse",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/SentencingMappingResource.kt:63]: seems odd this one is `409` when others are different?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:58]: yes - its just a made up code to identify this type of 409

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/SentencingMappingResource.kt (hunk 4 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/SentencingMappingResource.kt:63]: response code should be 409 and error code 1409

Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/SentencingMappingResource.kt:63]: I'll correct that error code that snuck in - should be 1409

**Pull request merged**